### PR TITLE
fix(test): Update 123done buttons to append flow params

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- Flow ids were not being appended to 123done client when clicking from subscription buttons

## This pull request

- Appends the flow ids
- Updates playwright tests to wait for the navigation to complete when coming from a subscribe button

## Issue that this pull request solves

Closes: # (issue number)

## Checklist

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/123done/static/js/123done.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ $(document).ready(function () {
   $.getJSON(
     `${paymentConfig.contentEnv}metrics-flow?form_type=button&utm_campaign=123done`
   ).done(function (data) {
-    $('.btn-subscribe-pwdless').each(function (index) {
+    $('.btn-subscribe').each(function (index) {
```
Comment (vbudhram) [inline packages/123done/static/js/123done.js:98]: This is only relevant change. The html above is updated for this, guess prettier hasn't touched the file.

File: packages/functional-tests/pages/relier.ts [hunk 2]
```diff
@@ -51,26 +51,32 @@ export class RelierPage extends BaseLayout {
...
     await this.page
       .getByRole('link', { name: 'Subscribe to Pro (USD)' })
       .click();
-    return waitForNavigation;
+    // Depending on if the user is signed in or not, match /products/ or /checkout/.
+    // We need to add a `waitUntil` option here because the page gets redirected from
+    // content-server to payments-server, and the URL changes before the page is fully loaded.
+    return this.page.waitForURL(/\/(checkout|products)\//, {
+      waitUntil: 'networkidle',
+    });
```
Comment (biancadanforth) [inline packages/functional-tests/pages/relier.ts:62]: Wouldn't this pattern match both the content server page URL and the payments server page URL for existing FxA users? I.e. we'd still only wait for the first navigation[1] and not the second?

[1]: sample content server URL from stage (note presence of **/products/**):

https://accounts.stage.mozaws.net/subscriptions/products/prod_FiJ42WCzZNRSbS?plan=price_1J0Y1iKb9q6OnNsLXwdOFgDr...

[2]: sample payments server page URL for existing FxA users (note presence of **/products/**):

https://payments-stage.fxa.nonprod.cloudops.mozgcp.net/products/prod_FiJ42WCzZNRSbS?plan=price_1J0Y1iKb9q6OnNsLXwdOFgDr...

**Review by biancadanforth — COMMENTED** [review]
Hey Vijay; thanks for working on this! I had a question for you as I'm wondering if this fix will work for existing FxA users that are logged in on the redirect?

**Comment by ashrivastava-qa** [comment]
@vbudhram looks like the new change for waitForUrl() is breaking the 'payapal' tests in the CI as it's unable to navigate to the URL mentioned.

**Comment by vbudhram** [comment]
Closing in favor of https://github.com/mozilla/fxa/pull/15196

**Pull request closed without merging**